pkg/discovery/systemd.go

```diff
@@ -107,3 +108,125 @@ func (c *SystemdDiscoverer) Run(ctx context.Context, up chan<- []*Group) error {
 		}
 	}
 }
+
+type Systemd2Config struct{}
+
+func NewSystemd2Config() *Systemd2Config {
+	return &Systemd2Config{}
+}
+
+func (c *Systemd2Config) Name() string {
+	return "systemd2"
+}
+
+func (c *Systemd2Config) NewDiscoverer(d DiscovererOptions) (Discoverer, error) {
+	return &Systemd2Discoverer{
+		logger: d.Logger,
+		prev:   make(map[string]systemd2.Unit),
+	}, nil
+}
+
+type Systemd2Discoverer struct {
+	logger log.Logger
+	prev   map[string]systemd2.Unit
+}
+
+func (d *Systemd2Discoverer) Run(ctx context.Context, up chan<- []*Group) error {
```

Comment:
I've tried to closely follow the existing implementation of `Run` and `SubscribeUnitsCustom` to make sure I haven't introduced a new behavior.